source/dol/bases/d_wm_enemy.cpp

```diff
@@ -0,0 +1,657 @@
+#include <game/bases/d_wm_enemy.hpp>
+#include <game/bases/d_wm_lib.hpp>
+#include <game/bases/d_wm_se_manager.hpp>
+#include <game/bases/d_wm_effect_manager.hpp>
+#include <game/bases/d_a_wm_map.hpp>
+#include <game/bases/d_a_wm_player.hpp>
+#include <game/bases/d_s_world_map_static.hpp>
+#include <game/bases/d_cs_seq_manager.hpp>
+#include <game/bases/d_game_com.hpp>
+#include <game/bases/d_info.hpp>
+
+const char *dWmEnemy_c::smc_PathPointA[] = {
+    "Fa00", "Fa01", "Fa02", "Fa03", "Fa04",
+    "Fa05", "Fa06", "Fa07", "Fa08"
+};
+const char *dWmEnemy_c::smc_PathPointB[] = {
+    "Fb00", "Fb01", "Fb02", "Fb03", "Fb04",
+    "Fb05", "Fb06", "Fb07", "Fb08"
+};
+const char *dWmEnemy_c::smc_PathPointC[] = {
+    "Fc00", "Fc01", "Fc02", "Fc03", "Fc04",
+    "Fc05", "Fc06", "Fc07", "Fc08"
+};
+
+namespace dWmEnemy {
+
+const ProfileName sc_EnemyProfName[] = {
+    fProfile::WM_KURIBO,
+    fProfile::WM_BROS,
+    fProfile::WM_PUKU,
+    fProfile::WM_PAKKUN,
+    fProfile::WM_JUGEM
+};
+
+bool isEnemyWalk() {
+    int courseType = dWmLib::GetCourseTypeFromCourseNo(dInfo_c::m_instance->m_54);
+    if (dScWMap_c::m_WorldNo == dScWMap_c::m_PrevWorldNo &&
+        dScWMap_c::m_SceneNo == dScWMap_c::m_PrevSceneNo &&
+        dWmLib::GetCurrentPlayResultStatus() != 0 &&
+        dWmLib::GetCurrentPlayResultStatus() != 9 &&
+        courseType != dWmLib::COURSE_TYPE_KINOKO_START &&
+        courseType != dWmLib::COURSE_TYPE_PEACH_CASTLE &&
+        courseType != dWmLib::COURSE_TYPE_INVALID &&
+        courseType != dWmLib::COURSE_TYPE_KINOKO &&
+        courseType != dWmLib::COURSE_TYPE_CANNON
+    ) {
+        return true;
+    }
+    return false;
+}
+
+} // namespace dWmEnemy
+
+dWmEnemy_c::~dWmEnemy_c() {
+    if (mpBgmSync != nullptr) {
+        delete mpBgmSync;
+    }
+}
+
+void dWmEnemy_c::initializeBase(const char **names, int count, bool cyclic) {
+    daWmMap_c *wmMap = daWmMap_c::m_instance;
+    dInfo_c::enemy_s enData;
+    dInfo_c::m_instance->GetMapEnemyInfo(dScWMap_c::m_WorldNo, mParam & 0xf, enData);
```

Comment:
Possible to define a mask for the enemy id extracted from mParam? We should decide how to handle mParam for actors in the future...